Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -137,6 +138,21 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               M_xface(i, j, k, 2) += dt * (-PhysConst::mu0 * mag_gamma_interp) * ( M_xface(i, j, k, 0) * Hy_eff - M_xface(i, j, k, 1) * Hx_eff)
                 + dt * Gil_damp * ( M_xface(i, j, k, 0) * ( M_xface(i, j, k, 2) * Hx_eff - M_xface(i, j, k, 0) * Hz_eff)
                 - M_xface(i, j, k, 1) * ( M_xface(i, j, k, 1) * Hz_eff - M_xface(i, j, k, 2) * Hy_eff));
+
+          // temporary normalized magnitude of M_xface field at the fixed point
+              Real M_normalized = sqrt( pow(M_xface(i, j, k, 0),2.0) + pow(M_xface(i, j, k, 1),2.0) + pow(M_xface(i, j, k, 2),2.0) ) /
+                                  MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+
+              // check the normalized error
+              if (M_normalized < (1.0-M_normalized_error) || M_normalized > (1.0+M_normalized_error)){
+                  amrex::Print() << "i " << i << " j " << j << " k " << k << "\n";
+                  amrex::Abort("Exceed the normalized error of the M_xface field");
```

Comment:
amrex::Print does not work within a GPU loop (amrex::ParallelFor)
you must use std::printf().  look up the calling sequence for this with variable arguments and let me know if you have trouble.

Reply:
Now use printf();